lib/samples/show_legend/show_legend_sample.dart

```diff
@@ -0,0 +1,175 @@
+            ),
+            // Add a dropdown button to the widget tree.
+            DropdownButtonHideUnderline(
+              child: DropdownButton( 
+                menuMaxHeight: 200,
+                alignment: Alignment.center,
+                hint: const Text(
+                  'Legend',
+                  style: TextStyle(
+                    color: Colors.deepPurple,
+                  ),
+                ),
+                elevation: 16,
+                style: const TextStyle(color: Colors.deepPurple),
+                // No need to set up onChanged callback.
+                onChanged: (_) {},
+                items: _legendsDropDown,
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  void onMapViewReady() async {
+    // Get the screen scale.
+    final screenScale = MediaQuery.of(context).devicePixelRatio;
+    // Create an image layer.
+    final imageLayer = ArcGISMapImageLayer.withUri(
+      Uri.parse(
+          'https://sampleserver6.arcgisonline.com/arcgis/rest/services/Census/MapServer'),
+    );
+    // Create a feature table.
+    final featureTable = ServiceFeatureTable.withUri(
+      Uri.parse(
+          'https://sampleserver6.arcgisonline.com/arcgis/rest/services/Recreation/FeatureServer/0'),
+    );
+    // Create a feature layer.
+    final featureLayer = FeatureLayer.withFeatureTable(featureTable);
+    // Add image and feature layers to the operational layers list of the map.
+    _map
+      ..operationalLayers.add(imageLayer)
+      ..operationalLayers.add(featureLayer);
+    // Load the image and feature layers.
+    await featureLayer.load();
+    await imageLayer.load();
+    // Create a list to store operational layers and populate it with image and feature layers.
+    final operationalLayersList = <LayerContent>[
+      ...imageLayer.subLayerContents,
+      featureLayer
+    ];
+
+    // Get the legend info for each layer and add it to the legends dropdown list.
+    for (final layer in operationalLayersList) {
+      // Get the legend info for the current layer.
+      final layerLegends = await layer.fetchLegendInfos();
+      // Add the name of the current layer.
+      _legendsDropDown.add(
+        DropdownMenuItem(
+          value: layerLegends.first,
+          child: Text(
+            layer.name,
+            style: const TextStyle(
+              fontSize: 15,
+              fontWeight: FontWeight.bold,
+            ),
+          ),
+        ),
+      );
+
+      // Add current layer's legends to the dropdown list.
+      for (final legend in layerLegends) {
+        final ArcGISImage? arcGISImage;
+        // Create a swatch for the legend if the legend exists.
+        if (legend.symbol != null) {
+          arcGISImage = await legend.symbol!.createSwatch(
+            screenScale: screenScale,
+            backgroundColor: Colors.transparent,
+            size: const Size.square(6),
+          );
+        } else {
+          arcGISImage = null;
```

Comment:
I think you don't need this `else` block. `arcGISImage` defaults to `null` automatically.

Reply:
If I remove it, it says `The final variable 'arcGISImage' can't be read because it's potentially unassigned at this point.
Ensure that it is assigned on necessary execution paths.`

Reply:
Oh, maybe that's because it's declared as `final ArcGISImage? arcGISImage`. What happens if you declare it as just `ArcGISImage? arcGISImage`

Reply:
That works!